src/System.Linq/src/System/Linq/Set.cs

```diff
@@ -10,17 +9,21 @@ namespace System.Linq
 {
     internal class Set<TElement>
     {
+        private readonly IEqualityComparer<TElement> _comparer;
         private int[] _buckets;
         private Slot[] _slots;
         private int _count;
-        private readonly IEqualityComparer<TElement> _comparer;
 #if DEBUG
         private bool _haveRemoved;
 #endif
 
         public Set(IEqualityComparer<TElement> comparer)
         {
-            if (comparer == null) comparer = EqualityComparer<TElement>.Default;
+            if (comparer == null)
+            {
+                comparer = EqualityComparer<TElement>.Default;
+            }
+
             _comparer = comparer;
```

Comment:
`_comparer = comparer ?? EqualityComparer<TElement>.Default;` better or worse? I prefer it, but it's mostly subjective. (There's a difference in IL produced too, but if it makes any difference to performance at all it'll be completely negligible).